From 0e022e4750517a47c34f64d78524a6eb84d708a6 Mon Sep 17 00:00:00 2001
From: Vladimir Smirnov <civil@gentoo.org>
Date: Mon, 19 May 2014 19:41:49 +0400
Subject: [PATCH 12/29] Set libperl soname

Bug-Gentoo: https://bugs.gentoo.org/286840

Patch-Name: gentoo/create_libperl_soname.diff

Ported to Perl 5.20
Ported to Perl 5.28.2 dilfridge
---
 Makefile.SH | 13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

diff --git a/Makefile.SH b/Makefile.SH
index 8774f291c8..d90b71bf66 100755
--- a/Makefile.SH
+++ b/Makefile.SH
@@ -70,11 +70,11 @@ true)
 				${revision}.${patchlevel}.${subversion}"
 		case "$osvers" in
 	        1[5-9]*|[2-9]*)
-			shrpldflags="$shrpldflags -install_name `pwd`/\$@ -Xlinker -headerpad_max_install_names"
+			shrpldflags="$shrpldflags -install_name `pwd`/libperl.${revision}.${patchlevel}.dylib -Xlinker -headerpad_max_install_names"
 			exeldflags="-Xlinker -headerpad_max_install_names"
 			;;
 		*)
-			shrpldflags="$shrpldflags -install_name \$(shrpdir)/\$@"
+			shrpldflags="$shrpldflags -install_name \$(shrpdir)/libperl.${revision}.${patchlevel}.dylib"
 			;;
 		esac
 		;;
@@ -84,13 +84,15 @@ true)
 		;;
 	sunos*)
 		linklibperl="-lperl"
+		shrpldflags="$shrpldflags -Wl,-soname -Wl,libperl.so.${revision}.${patchlevel}"
 		;;
-	netbsd*|freebsd[234]*|openbsd*|dragonfly*|bitrig*)
+	netbsd*|freebsd*|openbsd*|dragonfly*|bitrig*)
 		linklibperl="-L. -lperl"
+		shrpldflags="$shrpldflags -Wl,-soname -Wl,libperl.so.${revision}.${patchlevel}"
 		;;
 	interix*)
 		linklibperl="-L. -lperl"
-		shrpldflags="$shrpldflags -Wl,--image-base,0x57000000"
+		shrpldflags="$shrpldflags -Wl,--image-base,0x57000000 -Wl,-soname -Wl,libperl.so.${revision}.${patchlevel}"
 		;;
 	aix*)
 		case "$cc" in
@@ -128,6 +130,9 @@ true)
 	    linklibperl='libperl.x'
 	    DPERL_EXTERNAL_GLOB=''
 	    ;;
+	linux*)
+	    shrpldflags="$shrpldflags -Wl,-soname -Wl,libperl.so.${revision}.${patchlevel}"
+	    ;;
 	esac
 	case "$ldlibpthname" in
 	'') ;;
-- 
2.27.0

